activesupport/lib/active_support/current_attributes.rb

```diff
@@ -179,12 +179,7 @@ def current_instances_key
         end
 
         def method_missing(name, ...)
-          # Caches the method definition as a singleton method of the receiver.
-          #
-          # By letting #delegate handle it, we avoid an enclosure that'll capture args.
-          singleton_class.delegate name, to: :instance
-
-          send(name, ...)
+          instance.send(name, ...)
```

Comment:
```suggestion
          instance.public_send(name, ...)
```

Sorry, only noticed it now.